prow/plugins/reward-owners/reward-owners.go

```diff
@@ -0,0 +1,201 @@
+}
+
+func handle(ghc githubClient, gc git.ClientFactory, log *logrus.Entry, pr *github.PullRequest, info info, triggerConfig plugins.Trigger, skipTrustedUserCheck bool, resolver ownersconfig.Resolver) error {
+	org := info.org
+	repo := info.repo
+	number := info.number
+	filenames := resolver(org, repo)
+
+	// Get changes.
+	changes, err := ghc.GetPullRequestChanges(org, repo, number)
+	if err != nil {
+		return fmt.Errorf("error getting PR changes: %v", err)
+	}
+
+	// List added and removed users from OWNERS and OWNERS_ALIASES files.
+	addedUsers := sets.NewString()
+	removedUsers := sets.NewString()
+	for _, change := range changes {
+		if (filepath.Base(change.Filename) == filenames.Owners || filepath.Base(change.Filename) == filenames.OwnersAliases) && change.Status != github.PullRequestFileRemoved {
+			for _, s := range addedUserRe.FindAllStringSubmatch(change.Patch, -1) {
+				addedUsers.Insert(strings.TrimSpace(s[1]))
+			}
+			for _, s := range removedUserRe.FindAllStringSubmatch(change.Patch, -1) {
+				removedUsers.Insert(strings.TrimSpace(s[1]))
+			}
+		}
+	}
+
+	// Deduct removed from added users.
+	realAddedUsers := addedUsers.Difference(removedUsers)
+	if realAddedUsers.Len() == 0 {
+		return nil
+	}
+
+	// Check if the OWNERS_ALIASES file was modified.
+	var modifiedOwnerAliasesFile github.PullRequestChange
+	var ownerAliasesModified bool
+	for _, change := range changes {
+		if change.Filename == filenames.OwnersAliases {
+			modifiedOwnerAliasesFile = change
+			ownerAliasesModified = true
+			break
+		}
+	}
+
+	// Clone the repo, checkout the PR.
+	r, err := gc.ClientFor(org, repo)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.Config("user.name", "prow"); err != nil {
+		return err
+	}
+	if err := r.Config("user.email", "prow@localhost"); err != nil {
+		return err
+	}
+	if err := r.Config("commit.gpgsign", "false"); err != nil {
+		log.WithError(err).Errorf("Cannot set gpgsign=false in gitconfig: %v", err)
+	}
+	if err := r.MergeAndCheckout(pr.Base.Ref, string(github.MergeMerge), pr.Head.SHA); err != nil {
+		return err
+	}
+	// If OWNERS_ALIASES file exists, get all aliases.
+	_, _, repoAliases, err := verifyowners.NonTrustedUsersInOwnersAliases(ghc, log, triggerConfig, org, repo, r.Directory(), modifiedOwnerAliasesFile.Patch, ownerAliasesModified, skipTrustedUserCheck, filenames)
+	if err != nil {
+		return err
+	}
+
+	// Reward all users, unless it's an alias.
+	var errors []error
+	for _, user := range realAddedUsers.List() {
+		if _, exists := repoAliases[user]; exists {
+			continue
+		}
+		fmt.Println("let's reward", user)
+		if err := ghc.CreateComment(org, repo, number, RewardMessage); err != nil {
```

Comment:
We might make more of a coherent comment if we list the users, instead of posting (the same) information N times, once per user, in one PR.